api/v1beta1/glance_webhook.go

```diff
@@ -23,19 +23,51 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
+// GlanceDefaults -
+type GlanceDefaults struct {
+	ContainerImageURL string
+}
+
+var glanceDefaults GlanceDefaults
+
 // log is for logging in this package.
 var glancelog = logf.Log.WithName("glance-resource")
 
 // SetupWebhookWithManager sets up the webhook with the Manager
-func (r *Glance) SetupWebhookWithManager(mgr ctrl.Manager) error {
+func (r *Glance) SetupWebhookWithManager(mgr ctrl.Manager, defaults GlanceDefaults) error {
+	glanceDefaults = defaults
+
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(r).
 		Complete()
 }
 
-// TODO(user): EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+//+kubebuilder:webhook:path=/mutate-glance-openstack-org-v1beta1-glance,mutating=true,failurePolicy=fail,sideEffects=None,groups=glance.openstack.org,resources=glances,verbs=create;update,versions=v1beta1,name=mglance.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Defaulter = &Glance{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *Glance) Default() {
+	glancelog.Info("default", "name", r.Name)
+
+	r.Spec.Default()
+}
+
+// Default - set defaults for this Glance spec
+func (spec *GlanceSpec) Default() {
+	if spec.ContainerImage == "" {
+		spec.ContainerImage = glanceDefaults.ContainerImageURL
+	}
+
+	if spec.GlanceAPIExternal.ContainerImage == "" {
+		spec.GlanceAPIExternal.ContainerImage = glanceDefaults.ContainerImageURL
+	}
+
+	if spec.GlanceAPIInternal.ContainerImage == "" {
+		spec.GlanceAPIInternal.ContainerImage = glanceDefaults.ContainerImageURL
+	}
```

Comment:
Below @fmount, you ask about supporting different `ContainerImage` defaults for `GlanceAPIExternal` and `GlanceAPIInternal`. I think this code here needs to be changed to read different defaults from `glanceDefaults` to support that use case.

However if we only need to support manually defining different `ContainerImage` but the default is OK to be the same always. Then this code is OK as is.

Reply:
In this case `ContainerImage` is the same parameter that should be enforced to both `Internal` and `External`, so no big concerns. We can try refactoring the defaults in case we need to diverge a bit in the future.